### PR TITLE
Fix/additional allowance

### PR DESCRIPTION
**Pull request by prime623**
Check for necessary additional allowance on Create Swap when approving tokens.

### CODE REVIEW

File: backend/src/swaps/swaps.service.ts [hunk 1]
```diff
@@ -84,3 +85,39 @@ export const getFreeTrialEndTime = async () => {
 
   return { freeTrialEndTime: freeTrialEndTime.value };
 };
+
+export const getUserActiveSwaps = async (
```
Comment (GitMark0) [inline backend/src/swaps/swaps.service.ts:89]: Test this

File: backend/src/swaps/swaps.service.ts [hunk 2]
```diff
@@ -84,3 +85,39 @@ export const getFreeTrialEndTime = async () => {
...
+  seller: string,
+  tokenAddress: string,
```
Comment (GitMark0) [inline backend/src/swaps/swaps.service.ts:91]: Shouldn't this be optional? If a user wants to get all of his active swaps, regardless of the token address for example.

File: backend/src/swaps/swaps.service.ts [hunk 3]
```diff
@@ -84,3 +85,39 @@ export const getFreeTrialEndTime = async () => {
...
+        },
+      },
+      {
+        active: true,
+        chainId,
+      },
+    ],
+  }).select(["-__v", "-_id"]);
+
+  return { userActiveSwaps };
```
Comment (GitMark0) [inline backend/src/swaps/swaps.service.ts:122]: Why are we returning an object here?

File: backend/src/swaps/swaps.controller.ts [hunk 4]
```diff
@@ -51,3 +52,17 @@ export const getFreeTrialEndTime = async (req: Request, res: Response) => {
 
   return res.json(freeTrialEndTime);
 };
+
+export const getUserActiveSwaps = async (
+  req: GetUserActiveSwapsRequest,
+  res: Response
+) => {
+  const { seller, tokenAddress, chainId } = req.query;
+  const exists = await service.getUserActiveSwaps(
```
Comment (GitMark0) [inline backend/src/swaps/swaps.controller.ts:61]: This isn't really a good name for the return of `getUserActiveSwaps`.

**Review by GitMark0 — CHANGES_REQUESTED** [review]

File: backend/src/swaps/swaps.service.ts (hunk 3 above)
Comment (prime623) [reply backend/src/swaps/swaps.service.ts:122]: Is there a reason not to? I think it's more readable when the response is returned in this format, not just as an array of objects.

**Pull request merged**